data/crac/crac-result-extensions/src/main/java/com/farao_community/farao/data/crac_result_extensions/AbstractJsonResultExtension.java

```diff
@@ -12,23 +12,19 @@
 import com.fasterxml.jackson.core.JsonGenerator;
 import com.fasterxml.jackson.core.JsonParser;
 import com.fasterxml.jackson.core.type.TypeReference;
-import com.fasterxml.jackson.databind.DeserializationContext;
 import com.fasterxml.jackson.databind.SerializerProvider;
-import com.google.auto.service.AutoService;
 
 import java.io.IOException;
 import java.util.Map;
 
 /**
  * @author Baptiste Seguinot {@literal <baptiste.seguinot at rte-france.com>}
  */
-
-@AutoService(ExtensionsHandler.ExtensionSerializer.class)
-public class JsonResultExtension<T extends Identifiable<T>>
-    implements ExtensionsHandler.ExtensionSerializer<T, ResultExtension<T, ? extends Result>> {
+public abstract class AbstractJsonResultExtension<T extends Identifiable<T>, S extends AbstractResultExtension<T, ? extends Result>>
+    implements ExtensionsHandler.ExtensionSerializer<T, S> {
```

Comment:
Why not using directly ExtensionJsonSerializer from powsybl here ?

Reply:
Because this is defined in crac-impl and farao-crac-result-extension should not be dependent on crac-impl

Reply:
maybe we can move the ExtensionsHandler from crac-impl to crac-api instead?

Reply:
Yes indeed you can move it first to reverse the dependency ! But even after that I'm not sure this interface will be useful

Reply:
we use the same pattern as for other extension serializer (see for instance RaoComputationParameters or RaoComputationResult).

Maybe it has something to do with the `@autoservice` just above.

I'm not sure though it is usefull, we can try without it and implementing directly the JsonExtensionSerializer.

Reply:
So that it does not load all the JsonExtensionSerializer right ? In this case that's fine for me but maybe we should find a more explicit name CracResultExtensionSerializer or CracExtensionSerializer something like that

Reply:
i tried it an the test fails.
The subclasses of AbstractJsonResultExtension needs to implement `ExtensionsHandler.ExtensionSerializer` to be recognized by their `@autoservice`

Reply:
@TheMaskedTurtle I moved ExtensionsHandler from crac-impl to crac-api. Is it better?